### PR TITLE
Fix #1109 - multiple calendar instances with different config

### LINKED ISSUE #1109: Support multiple instances in Calendar
I'd like to create multiple calendars (a "Today's Calendar" with certain types of calendar items and a "Group Calendar" with upcoming events associated with a group of people, etc.).  I've added a new `instance` configuration option so you can create multiple calendar instances each have a unique instance id.  For example:

```
    {
      module: "calendar",
      header: "Today's Calendar",
      position: "top_left",
      config: {
        instance: "today",
        maximumNumberOfDays: 1,
        calendars: [
          {
            symbol: "cutlery ",
            url:
              "http://diningroomschedule.com"
          },
          {
            symbol: "male ",
            url:
              "https://mensevents.com"
          },
          {
            symbol: "female ",
            url:
              "https://womensevents.com"
          }
        ]
      }
    },
    {
      module: "calendar",
      header: "Group Calendar",
      position: "top_left",
      config: {
        instance: "bmoor",
        timeFormat: "absolute",
        getRelative: 0,
        urgency: 2,
        fadePoint: 0.7,
        dateFormat: "ddd, MMM do",
        excludedEvents: ["Breakfast", "Lunch"],
        calendars: [
          {
            symbol: "futbol-o ",
            maximumEntries: 2,
            url: "http://www.trumba.com/calendars/seattle-seahawks-schedule.ics"
          },
          {
            symbol: "futbol-o ",
            maximumEntries: 2,
            url: "http://collegefootballcalendar.net/cal/264"
          }
        ]
      }
    },

```

### PR DESCRIPTION
**Pull request by DarthBrento**
Fix issue #1109 
Allow multiple instances of calendar with different config by filtering the socket notifications by identifier.

### CODE REVIEW

File: modules/default/calendar/calendar.js [hunk 1]
```diff
@@ -122,6 +122,9 @@ Module.register("calendar", {
 
 	// Override socket notification handler.
 	socketNotificationReceived: function (notification, payload) {
+		if (this.identifier != payload.id)
```
Comment (rejas) [inline modules/default/calendar/calendar.js:125]: should be !== to check types too

Comment (DarthBrento) [reply modules/default/calendar/calendar.js]: fixed

Comment (rejas) [reply modules/default/calendar/calendar.js]: Thx. Also you should fix the tests, please run "npm run test:js" to see what can be fixed

Comment (DarthBrento) [reply modules/default/calendar/calendar.js]: Did it. 

The contribution guidelines mention `grunt`, but there is no `gruntfile` on the development branch. Am i supposed to do it by hand or is there any other tool to use?

Comment (rejas) [reply modules/default/calendar/calendar.js]: Grunt is removed on the develop branch so maybe we should remove it from the guidelines. Where is it mentioned exactly?

Comment (DarthBrento) [reply modules/default/calendar/calendar.js]: Here: https://github.com/MichMich/MagicMirror/blob/develop/.github/CONTRIBUTING.md

If you open a pull request for teh first time github points you there.

Comment (rejas) [reply modules/default/calendar/calendar.js]: Ah, right. That points to the master branch, which still mentions grunt. Once develop is merged with release 2.12 that will be fixed... Thanks for the PR and the tip.

**Comment by MichMich** [comment]
Is this ready to merge?

**Comment by DarthBrento** [comment]
Ready for merge!

**Comment by MichMich** [comment]
Thanks!

**Pull request merged**